File: client/src/types/report.ts
export type ReportStatus = "uploaded" | "processing" | "completed" | "failed";

export interface Report {
  id: string;
  user_id: string;
  file_name: string;
  file_type: string;
  file_size: number;
  storage_path: string;
  status: ReportStatus;
  uploaded_at: string;
}

File: server/app/models/report.py
from sqlalchemy import BigInteger, Column, DateTime, String, text
from sqlalchemy.dialects.postgresql import UUID

from app.db.session import Base


class Report(Base):
    __tablename__ = "reports"

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()"))
    user_id = Column(UUID(as_uuid=True), nullable=False, index=True)
    file_name = Column(String, nullable=False)
    file_type = Column(String, nullable=False)
    file_size = Column(BigInteger, nullable=False)
    storage_path = Column(String, nullable=False)
    status = Column(String, nullable=False, server_default="uploaded")
    uploaded_at = Column(DateTime(timezone=True), server_default=text("now()"))